src/axom/quest/MeshClipperStrategy.hpp

```diff
@@ -0,0 +1,377 @@
+
+  /*!
+   * @brief Get the 2D axis-aligned bounding box for the geometry,
+   * if it's applicable and available.
+  */
+  virtual const axom::primal::BoundingBox<double, 2>& getBoundingBox2D() const;
+
+  /*!
+   * @brief Get the 3D axis-alligned bounding box for the geometry,
+   * if it's applicable and available.
+  */
+  virtual const axom::primal::BoundingBox<double, 3>& getBoundingBox3D() const;
+
+  /*!
+   * @brief Label the cells in the mesh as inside, outside or
+   * both/undetermined, if possible.
+   *
+   * @param [in] shapeMesh Mesh to shape into.
+   * @param [out] labels Output
+   *
+   * The cell labels should be set to
+   * - @c labelIn if the cell is completely inside the shape,
+   * - @c labelOut if the cell is completely outside, and
+   * - @c labelOn if the cell is both inside and outside (or
+   *   cannot be easily determined).
+   *
+   * The output labels are used in optimizing the clipping algorithm.
+   * Subclasses should implementation this if it's cost-effective, and
+   * skip if it's not.  It's safe to label cells as on the boundary if
+   * it can't be possitively determined as inside or outside.
+   *
+   * @return Whether the operation was done.  (A false means
+   * not done.)
+   *
+   * If implemenation returns true, it should ensure these
+   * post-conditions hold:
+   * @post labels.size() == shapeMesh.getCellCount()
+   * @post labels.getAllocatorID() == shapeMesh.getAllocatorId()
+  */
+  virtual bool labelCellsInOut(quest::experimental::ShapeMesh& shapeMesh, axom::Array<LabelType>& cellLabels)
+  {
+    AXOM_UNUSED_VAR(shapeMesh);
+    AXOM_UNUSED_VAR(cellLabels);
+    return false;
+  }
+
+  /*!
+   * @brief Label the tetrahedra in certain cells, if possible.
+   *
+   * @param [in] shapeMesh Blueprint mesh to shape into.
+   * @param [in] cellIds Indices of cells whose constituent
+   *   tets should be labeled.
+   * @param [out] tetLabels Output
+   *
+   * Only the cells labeled as ON the boundary are subjected to
+   * this labeling.
+   *
+   * Tet indices refer to the @c shapeMesh.getCellsAsTets() array.
+   *
+   * If implemenation returns true, it should ensure these
+   * post-conditions hold:
+   * @post tetLabels.size() == TETS_PER_HEXAHEDRON * cellIds.size()
+   * @post labels.getAllocatorID() == shapeMesh.getAllocatorId()
+   * @post \c tetLabels should have \c TETS_PER_HEXAHEDRON labels
+   * for each index in \c cellIds.
+  */
+  virtual bool labelTetsInOut(quest::experimental::ShapeMesh& shapeMesh,
+                              axom::ArrayView<const axom::IndexType> cellIds,
+                              axom::Array<LabelType>& tetLabels)
+  {
+    AXOM_UNUSED_VAR(shapeMesh);
+    AXOM_UNUSED_VAR(cellIds);
+    AXOM_UNUSED_VAR(tetLabels);
+    return false;
+  }
+
+  /*!
+   * @brief Clip with a fast geometry-specialized method if
+   * possible.
+   *
+   * @param [in] shapeMesh Blueprint mesh to shape into.
+   * @param ovlap [out] Shape overlap volume of each cell
```

Comment:
Does this need to be changed?  I'm not sure.
```suggestion
   * @param [out] ovlap Shape overlap volume of each cell
```

Reply:
I think it does.